apis/mattermost/v1beta1/mattermost_types.go

```diff
@@ -117,6 +117,10 @@ type MattermostSpec struct {
 	// +optional
 	PodTemplate PodTemplate `json:"podTemplate,omitempty"`
 
+	// UpdateJob defines configuration for the template for the update job.
+	// +optional
+	UpdateJob UpdateJob `json:"updateJob,omitempty"`
```

Comment:
Lets make it a pointer to clearly indicate it is optional and avoid any unexpected issues in yaml because of that. I think we should make a `PodTemplate` a pointer as well it is generally a better practice since if we add some non-pointer field to the struct it may show up in yaml. No need to change it in this PR, just realized that we have missed it last time.

Reply:
I believe I resolved this and the other comments - I'll make a separate PR to change `PodTemplate` to a pointer just to keep things smaller and cleanly separate the changes.

Reply:
Resolving `podTemplate` comments with https://github.com/mattermost/mattermost-operator/pull/302 - wouldn't be surprised if there will be conflicts between the two once one is merged so I'll keep an eye on these and rebase as needed.

Reply:
Thanks for changing that as well, separate PR works just fine.